src/components/PasswordInput/__test__/PasswordInput.spec.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import { render, fireEvent, screen } from "@testing-library/react";
+import { PasswordInput } from "..";
+
+jest.mock('@zxcvbn-ts/core', () => ({
+    ...jest.requireActual('@zxcvbn-ts/core'),
+    zxcvbn: jest.fn(() => ({
+        score: 0,
+        feedback: {
+            warning: '',
+            suggestions: [],
+        },
+    })),
+}));
+
+describe("PasswordInput component", () => {
+    it("renders with default props", () => {
+        render(
+            <PasswordInput
+                hidePasswordMeter
+                hint="Password must be at least 8 characters"
+                label="Password"
+                isFullWidth={true} />
+        );
+        const input = screen.getByLabelText('Password');
+        expect(input).toBeInTheDocument();
+        expect(input).toHaveAttribute('type', 'password');
+        expect(screen.getByText('Password must be at least 8 characters')).toBeInTheDocument();
+    });
+
+    it("handles onChange event", () => {
+        const handleChange = jest.fn();
+        render(
+            <PasswordInput label="Password" onChange={handleChange} />
+        );
+        const input = screen.getByLabelText("Password");
+        fireEvent.change(input, { target: { value: "password123" } });
+        expect(handleChange).toHaveBeenCalledTimes(1);
+    });
```

Comment:
```suggestion
    it("handles onChange event", async () => {
        const handleChange = jest.fn();
        render(
            <PasswordInput label="Password" onChange={handleChange} />
        );
        const input = screen.getByLabelText("Password");
        await userEvent.type(input,"password123" );
        expect(handleChange).toHaveBeenCalledTimes(1);
    });
```
let's use `userEvent` since it simulates the real user interaction